cli/args/flags.rs

```diff
@@ -5727,6 +5750,8 @@ fn env_file_arg_parse(flags: &mut Flags, matches: &mut ArgMatches) {
   flags.env_file = matches
     .get_many::<String>("env-file")
     .map(|values| values.cloned().collect());
+
+  load_env_variables_from_env_file(flags.env_file.as_ref());
```

Comment:
I'm not 100% sure about that - this change means that any time we need to construct flags again (for whatever reason) we are going to implicitly load the `.env` file again. @dsherret wdyt about that?

Reply:
but flag parsing depends on environment variables. another example is `DENO_JOBS`.

Reply:
I don't really understand what you have in mind.

Reply:
I think we should leave clap for parsing cli args (not env vars) Three stages: arg parsing with clap, env file loading based on args, then at this point check the value of the env vars (could be backfilled into the flags struct or resolved in CliOptions). Some properties on the flags structs would need to be `None` to indicate no argument was provided instead of defaulting to a value. I can show an example tomorrow if that would help.